src/main/functions/Keys.ts

```diff
@@ -83,16 +67,16 @@ function getThumbprintFromJwkDIDKeyImpl(jwk: JWK): string {
   return `z${bs58.encode(buffer)}`;
 }
 
-export function getThumbprintFromJwk(jwk: JWK, did: string): string {
+export async function getThumbprintFromJwk(jwk: JWK, did: string): Promise<string> {
   if (isEd25519DidKeyMethod(did)) {
     return getThumbprintFromJwkDIDKeyImpl(jwk);
   } else {
-    return getThumbprintFromJwkImpl(jwk);
+    return await calculateJwkThumbprint(jwk, 'sha256');
```

Comment:
isn't it better to get it from the request or registration object?

Reply:
Not sure what you mean. Spec mandates a sha255 thumbprint and this replacement does exactly the same as the previous method it is replacing